src/devices/Bmp280/Bmp280.cs

```diff
@@ -0,0 +1,592 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+//ported from https://github.com/adafruit/Adafruit_BMP280_Library/blob/master/Adafruit_BMP280.cpp
+
+using System;
+using System.Device.I2c;
+using System.Device.I2c.Drivers;
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
+
+namespace Iot.Device
+{
+    public class Bmp280 : IDisposable, IBmp280
+    {
+        private I2cDevice _i2cDevice;
+        private CommunicationProtocol protocol;
+        private bool initialised = false;
+        private Bmp280CalibrationData CalibrationData;
+        protected readonly byte Signature = 0x58;
+        private int TFine;
+
+        private enum CommunicationProtocol
+        {
+            Gpio,
+            Spi,
+            I2c
+        }
+
+        public Bmp280(I2cDevice i2cDevice)
+        {
+            _i2cDevice = i2cDevice;
+            CalibrationData = new Bmp280CalibrationData();
+            protocol = CommunicationProtocol.I2c;
+        }
+
+        public Bmp280(I2cConnectionSettings settings)
```

Comment:
I would remove this constructor. I get that what you are trying to do is to make this work on Linux and Windows without changing the consuming code, but on past device bidnings we have opted to just provide the constructor that takes in the I2cDevice